.circleci/config.yml

```diff
@@ -101,146 +231,154 @@ jobs:
+            - ~/.gradle/wrapper
             - ~/.m2
             - ~/workbench/api/build/exploded-api/WEB-INF/lib/
-          key: api-cache-{{ checksum "~/workbench/api/build.gradle" }}
+          key: v1-api-cache-{{ checksum "~/workbench/api/build.gradle" }}
 
-  api-deploy-to-test:
-    <<: *java_defaults
+  restore_api_cache:
+    description: "workbench/api dir: restore gradle cache"
     steps:
-      - checkout
-      - run: *update_git_submodules
-      # Note: most of the time spent here appears to be in Gradle / App Engine
-      # deployment. We tried more aggressively caching outputs via Circle
-      # workspaces, but that seemed to have a negligible effect on speed. It's
-      # also tricky to pick specific sub directories since the API deploy
-      # touches several top level folders {common,}api.
       - restore_cache:
           keys:
-          - api-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-cache-
-      - run: *activate_service_accont_creds
-      - deploy:
-          name: Deploy to App Engine
+            - v1-api-cache-{{ checksum "~/workbench/api/build.gradle" }}
+            - v1-api-cache-
+
+# -------------------------
+#        JOBS
+# -------------------------
+jobs:
+  api-build-test:
+    executor: workbench_docker
+    steps:
+      - git_checkout
+      - ensure_branch_has_api_changes
+      - restore_api_cache
+      - run:
+          name: Validate swagger definitions
           working_directory: ~/workbench/api
-          command: |
-            ./project.rb deploy \
-              --project all-of-us-workbench-test \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
-              --version circle-ci-test \
-              --key-file circle-sa-key.json \
-              --promote
+          command: ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+      - run:
+          name: Integration tests compile
+          working_directory: ~/workbench/api
+          command: ./project.rb gradle compile__bigquerytest__Java compile__integration__Java
+      - run:
+          name: Run Java unit tests
+          working_directory: ~/workbench/api
+          command: ./project.rb test
+      - run:
+          # Lint last; it's more important to surface test failures early.
+          name: Java linting
+          working_directory: ~/workbench/api
+          command: ./gradlew spotlessCheck
+      - save_api_cache
+
+  api-local-test:
+    executor: mysql_docker
+    steps:
+      - start_api_local:
+          skip_no_api_change: false
+      - run:
+          name: Run API tests on running local server
+          working_directory: ~/workbench/api
+          command: ./project.rb run-local-api-tests
+      - persist_to_workspace:
+          root: .
+          paths:
+            - api
+
+  api-deploy-to-test:
+    executor: workbench_docker
+    steps:
+      # - git_checkout
+      # Use "api-local-test" job's workspace
```

Comment:
Interesting. I see how this would be faster, but I thought jobs weren't supposed to be order-dependent (though the steps in them are).